### ISSUE TITLE
[Planned] Character Stats

### ISSUE THREAD

**Issue by Adesii** [Adesii/IndieCure#6]
**Describe the Feature**
A Generic way to store stats on any Node or character. Similar to the GAS system in Unreal

Preferably any stat should be just stored as its own node. or if thats too much overhead it should be stored on a single node with a dictionary that gets handled by a generic Stat Singleton that allows getting and setting stats on any node

Also allowing to make default preset nodes that sets the stats on startup.

Should also Store Base Current
to allow modifications

**References**
Basically stats like Attack speed,HP etc.. from literally any game

**Comment by Adesii** [comment]
i think this is now implemented... might need modification later on to work better. but it works
